app/code/Magento/Catalog/Model/Product/Attribute/Repository.php

```diff
@@ -113,7 +113,11 @@ public function save(\Magento\Catalog\Api\Data\ProductAttributeInterface $attrib
                 ->getEntityType(\Magento\Catalog\Api\Data\ProductAttributeInterface::ENTITY_TYPE_CODE)
                 ->getId()
         );
-        if ($attribute->getAttributeId()) {
+        $existingAttributeModel = $this->eavConfig->getAttribute(
```

Comment:
Hi @somat211. Thank you for your collaboration. 
From the performance perspective, we should execute this action only in case when `attribute_id` is missing. Otherwise, we don't need an additional database request.

Reply:
Hi @rogyar. Thank you for your feedback. I have updated the code as per your suggestion. Could you please review it.